src/js/Gameboard.js

```diff
@@ -0,0 +1,105 @@
+import {Ship, allShips} from "./Ship";
+
+const Gameboard = () => {
+  let board = new Array(100).fill(0);
+  const COLUMN = 10;
+  const SLOT = {
+    EMPTY: 0,
+    MISS: 1,
+    HIT: 6
+  };
+
+  const randomPlacement = () => {
+
+    const direction = () => {
+      let hv = ['horizontal', 'vertical'];
+      let direction = Math.floor(Math.random() * hv.length);
+      return hv[direction];
+    };
+
+    const getShipsName = () => {
+      let shipName = [];
+      for (let ship in allShips) {
+        shipName.push(ship);
+      }
+      return shipName;
+    };
+
+    const getShipsSize = () => {
+      let shipSize = [];
+      for (let ship in allShips) {
+        shipSize.push(allShips[ship].size);
+      }
+      return shipSize;
+    };
+
+    const horOverlap = (board, boardNum, size, i) => {
+      let sizeCount = 0;
+      for (let j = boardNum; j < boardNum + size[i]; j++) {
+        if (board[j] === SLOT.EMPTY){
+          sizeCount += 1;
+        }
+      }
+      return sizeCount === size[i];
+    };
+
+    const verOverlap = (board, boardNum, size, i) => {
+      let sizeCount = 0;
+      for (let j = boardNum; j < boardNum + size[i] * COLUMN; j += 10) {
+        if (board[j] === SLOT.EMPTY){
+          sizeCount += 1;
+        }
+      }
+      return sizeCount === size[i];
+    };
```

Comment:
horOverlap and verOverlap are almost identical. Try to get them into a single method to avoid repetition

Reply:
It's not identical due to the increment step